skytemple_randomizer/randomizer/npc.py

```diff
@@ -91,6 +107,167 @@ def run(self, status: Status):
+                "Spanish": re.compile(r"Tienda\n?(Kecleon)"),
+                "Japanese": re.compile(r"(カクレオン)(?:\[CR])?の\n?お?みせ"),
+            }
+            kecleon_replace_regions = [
+                self.static_data.string_index_data.string_blocks.get(
+                    "Floor-Wide Status Names+Desc"
+                ),
+                self.static_data.string_index_data.string_blocks.get(
+                    "Item Long Descriptions"
+                ),
+            ]
+            # Finally, there are plain texts that need replacing for a bunch of Pokemons (typically chapter texts and place names)
+            plain_npc_text = re.compile(
+                "(" + "|".join(list(mapped_actor_names.keys())) + ")"
+            )
+            plain_replace_regions = [
+                self.static_data.string_index_data.string_blocks.get(
+                    "(SPECIAL-EPISODES-STRS-UNMAPPED)"
+                ),
+                self.static_data.string_index_data.string_blocks.get(
+                    "(JOURNAL-STRS-UNMAPPED)"
+                ),
+                self.static_data.string_index_data.string_blocks.get(
+                    "Pokemon WAIT Dialogue"
+                ),
+                self.static_data.string_index_data.string_blocks.get(
+                    "Ground Map Names"
+                ),
+                self.static_data.string_index_data.string_blocks.get(
+                    "Dungeon Names (Main)"
+                ),
+                self.static_data.string_index_data.string_blocks.get(
+                    "Dungeon Names (Section)"
+                ),
+            ]
+
+            for idx, text in enumerate(lang_string_file.strings):
+                new_text = standard_npc_text.sub(
+                    lambda match: match.expand(
+                        f"[CS:{match.group(1)}]{mapped_actor_names[match.group(2)]}[CR]"
+                    ),
+                    text,
+                )
+                new_text = place_mention_npc_text.sub(
+                    lambda match: match.expand(
+                        f"[CS:P]{match.group(1)}{mapped_actor_names[match.group(2)]}{match.group(3)}[CR]"
+                    ),
+                    new_text,
+                )
+                if any(
+                    block.begin < idx < block.end
+                    for block in csk_replace_regions
+                    if block is not None
+                ):
+                    new_text = csk_npc_text.sub(
+                        lambda match: match.expand(
+                            f"[CS:K]{mapped_actor_names[match.group(1)]}[CR]"
+                        ),
+                        new_text,
+                    )
+                if any(
+                    block.begin < idx < block.end
+                    for block in kecleon_replace_regions
+                    if block is not None
+                ):
+                    new_text = kecleon_shop_text[lang.name_localized].sub(
```

Comment:
This needs to be `lang.name`, otherwise it breaks with non-english UI localizations. I'll fix that in my next commit.